src/solvers/strings/string_constraint_generator_main.cpp

```diff
@@ -441,9 +441,12 @@ exprt zero_if_negative(const exprt &expr)
 /// the return codes and merging the constraints.
 std::pair<exprt, string_constraintst> combine_results(
   std::pair<exprt, string_constraintst> result1,
-  std::pair<exprt, string_constraintst> result2)
+  std::pair<exprt, string_constraintst> result2,
+  const namespacet &ns)
 {
-  const exprt return_code = maximum(result1.first, result2.first);
+  exprt return_code = maximum(result1.first, result2.first);
   merge(result2.second, std::move(result1.second));
+  simplify(return_code, ns);
```

Comment:
the change should be documented

Reply:
Anywhere specific?

Reply:
in the documentation of `combine_results`